core/lib/slider.ts

```diff
@@ -0,0 +1,569 @@
+import {computed, writable} from '@amadeus-it-group/tansu';
+import type {WidgetsCommonPropsAndState} from './commonProps';
+import type {ConfigValidator, PropsConfig} from './services';
+import {bindableDerived, registrationArray, writablesForProps} from './services';
+import type {ReadableSignals} from './services/stores';
+import {mergeConfigStores, normalizeConfigStores, stateStores} from './services/stores';
+import {typeArray, typeBoolean, typeFunction, typeNumber, typeString} from './services/writables';
+import type {Directive, Widget} from './types';
+import {noop} from './utils';
+
+export interface SliderCommonPropsAndState extends WidgetsCommonPropsAndState {
+	/**
+	 * Minimum value that can be assigned to the slider
+	 */
+	min: number;
+
+	/**
+	 * Maximum value that can be assigned to the slider
+	 */
+	max: number;
+
+	/**
+	 * Unit value between slider steps
+	 */
+	stepSize: number;
+
+	/**
+	 * If `true` slider value cannot be changed but the slider is still focusable
+	 */
+	readonly: boolean;
+
+	/**
+	 * If `true` slider value cannot be changed and the slider cannot be focused
+	 */
+	disabled: boolean;
+
+	/**
+	 * If `true` is vertically positioned otherwise it is horizontal
+	 */
+	vertical: boolean;
+
+	/**
+	 * Dom Rectangle of the slider
+	 */
+	sliderDomRect: DOMRect;
+
+	/**
+	 * Dom Rectangle of the minimum label
+	 */
+	minLabelDomRect: DOMRect;
+
+	/**
+	 * Dom Rectangle of the maximum label
+	 */
+	maxLabelDomRect: DOMRect;
```

Comment:
Do you use `sliderDomRect`, `minLabelDomRect` or `maxLabelDomRect` outside of the core?
As far as I can see, it is not useful to expose them in the state, they could be part of `SliderProps` instead of `SliderCommonPropsAndState`.

Reply:
Actually, instead of using props, maybe it would be better to have directives for each of those DOM elements, and then observing the size of the elements could be done in the core, and that would avoid repeating this logic in other frameworks.

Reply:
It would be nice to avoid duplicating resize observer logic across frameworks indeed

Reply:
But then again we could see it when implementing React or Svelte version?

Reply:
agree, it is not used outside the core, only updated from there, it should be only a prop

Reply:
Actually, as I said in my second comment, it should not even be a prop, because it does not really make sense to correspond to an `@Input` in the angular component. So having an internal writable in the core that is updated by an exposed directive that is put on the corresponding DOM elements seems more appropriate to me.